### DISCUSSION TITLE
How to build custom model and train on my dataset?

### DISCUSSION THREAD

**Discussion by ACALJJ32** [open-mmlab/mmediting#1269] [Datasets] [answered]
Hi,I'm glad to use mmediting,but I don't know how to define my model and train(test) on my dataset,could your please teach me how to modify your tool in mmediting?

**Comment by innerlee** [comment]
Hi, thanks for trying mmedit! For a specific task, you may prepare your data similar to an exisiting one, and modify the data path in existing configs (here https://github.com/open-mmlab/mmediting/tree/master/configs) to your own data path.

If the data format is good, there will be no code change required.

**Comment by ACALJJ32** [comment]
Thank you for your reply! But I still have a question: When I want to use basicVSR model to test on REDS4 dataset, I still confused about where did the model find the image : 000/00000000.png.  It said that "It reads REDS keys from the txt file. Each line contains:
    1. image name; 2, image shape, seperated by a white space.
    Examples:
        000/00000000.png (720, 1280, 3)
        000/00000001.png (720, 1280, 3)", 
but I don't know where I can foud the txt file.

**Comment by wwhio** [comment]
You need to prepare reds dataset by [script](https://github.com/open-mmlab/mmediting/tree/master/tools/data/super-resolution/reds)

**Comment by Yshuo-Li** [comment]
> Hi,I'm glad to use mmediting,but I don't know how to define my model and train(test) on my dataset,could your please teach me how to modify your tool in mmediting?

** how to define my model **

Maybe you can define your model refer to existing models.
For example, if your model is a single end-to-end model, you can define it refer to EDSR (https://github.com/open-mmlab/mmediting/blob/master/mmedit/models/backbones/sr_backbones/edsr.py).
Otherwise, if you want to define train-step and valid-step, BasicVSR and EDVR are good examples.

**Comment by ACALJJ32** [comment]
I have tested the model and realized where to define the folder :['000', '011', '015', '020']，but I still don't understand how did the model find the image path: 000/00000000.png, it must be defined in somewhere. And in the configs/restorers/basivsr_reds4.py, it's seems like we don't need to define the ann_file in pipeline.

**Comment by ckkelvinchan** [comment, marked as answer]
Hi @ACALJJ32, it is defined [here](https://github.com/open-mmlab/mmediting/blob/master/mmedit/datasets/pipelines/augmentation.py#L932). If the argument `num_input_frames` is not provided, `GenerateSegmentIndices` automatically extracts the entire sequence (which is equivalent to starting from `000000000.png` and extracting 100 frames for REDS.)

**Comment by ckkelvinchan** [comment]
Moreover, for BasicVSR on REDS, you should look for [SRREDSMultipleGTDataset](https://github.com/open-mmlab/mmediting/blob/master/mmedit/datasets/sr_reds_multiple_gt_dataset.py).

**Comment by ACALJJ32** [comment]
Thank you! I think I have understand  how to continue my work.
